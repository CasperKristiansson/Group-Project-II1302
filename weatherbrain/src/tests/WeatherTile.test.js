/*Philip Hägg */

import React from 'react';
import ReactDom from 'react-dom';
import WeatherTile from "../components/weathertile.js";
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';


afterEach(cleanup);
it ("renders withouth crashing", () => {
    const div = document.createElement('div');
    ReactDom.render(<WeatherTile />, div)
    ReactDom.unmountComponentAtNode(div);
})

it("renders weather tile correctly with props, date", () => {
    const { getByText } = render(<WeatherTile date="2020-01-01" icon="sunny" temperature="22°C" />);
    const date = getByText("2020-01-01");
    expect(date).toBeInTheDocument("2020-01-01");
})



/*SNAP SHOT TEST */
it("matches snapshot", () => {
    const tree = renderer.create(<WeatherTile date="2020-01-01" icon="sunny" temperature="22°C" />).toJSON();
    expect(tree).toMatchSnapshot();
})
