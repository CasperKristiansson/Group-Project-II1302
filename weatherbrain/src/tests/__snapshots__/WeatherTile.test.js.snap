// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="tilediv"
>
  <div
    class="content-holder"
  >
    <div
      class="date"
    >
      2020-01-01
    </div>
    <div>
      <div
        class="icon-holder"
      >
        <link
          href="https://fonts.sandbox.google.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0"
          rel="stylesheet"
        />
        <span
          class="material-symbols-outlined"
        >
          sunny
        </span>
      </div>
      <div
        class="temperature-holder"
      >
        22Â°C
      </div>
    </div>
  </div>
</div>
`;
